### ISSUE TITLE
Setting to prevent Zed from changing indentation of current line while I type

### ISSUE THREAD

**Issue by aethertap** [zed-industries/zed#15533]
### Check for existing issues

- [X] Completed

### Describe the feature

While typing, there are certain characters that cause zed to reindent the line while I'm working on it. This is frustrating when I'm working on a multi-language file or trying to adhere to a particular style. It would be great if this behavior could be disabled with a setting change.

This is related to [#11780](https://github.com/zed-industries/zed/issues/11780), but in this case the problem is on the current line rather than on the newly-created one.

**Comment by aethertap** [comment]
It still happens, here is my settings.json in case I put it in the wrong place:

It's happening within a proc macro in a leptos component, in case that is important here. 

```json
{
  "theme": "Summercamp",
  "vim_mode": true,
  "ui_font_size": 16,
  "buffer_font_size": 14,
  "format_on_save": "off",
  "use_on_type_format":false,
  "lsp": {
    "rust-analyzer": {
      "procMacro": {
        "ignored": [
          "server"
        ]
      }
    }
  }
}
````

**Comment by SomeoneToIgnore** [comment]
Yeah, it's not very clear what happens, it would be helpful to get a repro example (a project with the set of instructions and/or the video)

**Comment by aethertap** [comment]
Working on an example, but here's the video. It reindents the line I'm working on when I hit the `=` key, which is what I would like to prevent if possible.

[Video](https://github.com/user-attachments/assets/f1bfb6b2-acc8-4b31-8605-bc9b94451467)

**Comment by aethertap** [comment]
It was actually hard to get a minimum example to reproduce the problem, so it seems to be specific to this particular setup, and specifically within the `#[component]` functions. [Here's a repo](https://github.com/aethertap/zed-indent). Sorry it's an entire repo, I tried to get it down to just a file, but I couldn't make it happen. If you edit `src/app.rs`, try inserting `class=` under the line with the button; that reliably reindents the line for me.

**Comment by SomeoneToIgnore** [comment]
Thank you, no worries about the minimal example, great that you have something to try on my side — I'm rarely getting any repro projects shared at all, unfortunately.

**Comment by SomeoneToIgnore** [comment]
Interesting, no formatting appears in the LSP logs at all, and the only LSP server that works there is rust-analyzer.

Which means, this is a language feature, and most probably this is the 
https://github.com/zed-industries/zed/pull/13351/files
one, as every language that does not have `auto_indent_using_last_non_empty_line` have it set to `true`

You can verity this hypothesis very easily by aligning the line above somehow differently — you'll see a different "formatting" in play.

I am not sure what is a good way to solve this — disabling the whole feature for Rust seems bad, as there are many places where it is useful.
I am not sure also that it's possible to disable this feature in runtime.

Feels that we miss the fact that Rust may have some extra, inner languages used and need to somehow adjust the language settings for that?
Unfortunately, I am not sure what to do here, but at least you know the workaround for now: it's the line above the "formatting" will align the new line to.

**Comment by aethertap** [comment]
Okay, that makes sense. Thanks for looking into it, I will work around it for now.

**Comment by pbsds** [comment]
Also happens with multi-cursor

https://github.com/user-attachments/assets/62eb53b2-b032-444f-80e1-a668ce926a5c

**Comment by bsgbryan** [comment]
Another example; in Rust:

https://github.com/user-attachments/assets/d3fdba53-1f8e-4954-a9a0-532eff0cc849

Admittedly, I have... unique... code formatting styles - but I really love Zed and don't want to fight with this kind for formatting stuff.

**Comment by TrueSgtMonkey** [comment]
This is a very annoying issue and should be looked into. Makes the editor unusable

**Comment by CarlosUriass** [comment]
Reproducible on macOS Stable; Zed automatically reindents the current line when typing }.

Environment

OS: macOS 26.5.1 (arm64)
Zed: 1.14.2+stable.337.02abf5b08fa12c1c20a155ae3f796ef4c6c1a01e (Stable)

Steps

1. Created a test.js file with the following contents:

```
if (true) {
	console.log("test");
}
```

2. Placed the cursor at the end of the first line, after {, and pressed Enter.
3. Typed foo, producing:

```
if (true) {
	foo
	console.log("test");
}
```

4. Moved the cursor to the beginning of the foo line.
5. Typed } before foo.

Result

Reproducible on my end.

When } is typed at the beginning of the current line, Zed automatically reindents the line while it is being edited. The resulting line becomes:

`}foo`

with the indentation being adjusted automatically.

This matches the issue description: the current line is reindented as a result of typing a character, rather than only applying indentation when creating a new line.

@yeskunall @AidanV

<img width="368" height="160" alt="Image" src="https://github.com/user-attachments/assets/288f93f5-fe58-4d1b-9b18-6bbdf937a9e2" />
